arches/app/etl_modules/bulk_data_deletion.py

```diff
@@ -129,19 +130,34 @@ def get_sample_data(self, nodegroup_id, resourceids):
 
         return sample_data[0:5]
 
-    def delete_resources(self, userid, loadid, graphid, resourceids):
+    def delete_resources(self, userid, loadid, graphid, resourceids, verbose=False):
         result = {"success": False}
-        user = User.objects.get(id=userid)
+        deleted_count = 0
+        try:
+            user = User.objects.get(id=userid)
+        except ValueError:
+            user = None
         try:
             if resourceids and graphid:
                 resources = Resource.objects.filter(graph_id=graphid).filter(pk__in=resourceids)
             elif graphid:
                 resources = Resource.objects.filter(graph_id=graphid)
             elif resourceids:
                 resources = Resource.objects.filter(pk__in=resourceids)
+            if verbose is True:
+                deleted_count = resources.count()
+                bar = pyprind.ProgBar(deleted_count)
             for resource in resources.iterator(chunk_size=2000):
                 resource.delete(user=user, index=False, transaction_id=loadid)
+                if verbose is True:
+                    bar.update()
+                else:
+                    deleted_count += 1
```

Comment:
I'm wondering if it's really necessary to track the deleted_count. Couldn't resources.count() be used whether or not verbose is True?

Reply:
fix in latest